packages/aws-rfdk/lib/lambdas/nodejs/x509-certificate/test/acm-handlers.test.ts

```diff
@@ -0,0 +1,459 @@
+      // An additional check is made before logging "Retrying..."
+      expect(shouldContinueStub.callCount).toEqual(4);
+    });
+
+    describe('existing', () => {
+      test('throws if item ARN is missing', async () => {
+        // GIVEN
+        const resourceTable = new MockCompositeStringIndexTable();
+        const getItemStub = sinon.stub(resourceTable, 'getItem').resolves({});
+
+        const importer = new TestAcmCertificateImporter({
+          acm: new AWS.ACM(),
+          dynamoDb: new AWS.DynamoDB(),
+          secretsManager: new AWS.SecretsManager(),
+          resourceTableOverride: resourceTable,
+        });
+
+        // WHEN
+        await expect(importer.doCreate(physicalId, doCreateProps))
+
+        // THEN
+          .rejects.toEqual(new Error("Database Item missing 'ARN' attribute"));
+        expect(getItemStub.calledOnce).toBe(true);
+      });
+
+      test('throws if certificate not found in ACM', async () => {
+        // GIVEN
+        const resourceTable = new MockCompositeStringIndexTable();
+        const getItemStub = sinon.stub(resourceTable, 'getItem').resolves({ ARN: certArn });
+
+        const getCertificateFake = sinon.fake.resolves({});
+        AWSMock.mock('ACM', 'getCertificate', getCertificateFake);
+
+        const importer = new TestAcmCertificateImporter({
+          acm: new AWS.ACM(),
+          dynamoDb: new AWS.DynamoDB(),
+          secretsManager: new AWS.SecretsManager(),
+          resourceTableOverride: resourceTable,
+        });
+
+        // WHEN
+        await expect(importer.doCreate(physicalId, doCreateProps))
+
+        // THEN
+          .rejects.toEqual(new Error(`Database entry ${certArn} could not be found in ACM.`));
+        expect(getItemStub.calledOnce).toBe(true);
+        expect(getCertificateFake.calledOnce).toBe(true);
+      });
+
+      test('imports certificate', async () => {
+        // GIVEN
+        const resourceTable = new MockCompositeStringIndexTable();
+        const getItemStub = sinon.stub(resourceTable, 'getItem').resolves({ ARN: certArn });
+
+        const getCertificateFake = sinon.fake.resolves({ Certificate: 'cert' });
+        AWSMock.mock('ACM', 'getCertificate', getCertificateFake);
+
+        const importCertificateFake = sinon.fake.resolves({});
+        AWSMock.mock('ACM', 'importCertificate', importCertificateFake);
+
+        const importer = new TestAcmCertificateImporter({
+          acm: new AWS.ACM(),
+          dynamoDb: new AWS.DynamoDB(),
+          secretsManager: new AWS.SecretsManager(),
+          resourceTableOverride: resourceTable,
+        });
+
+        // WHEN
+        await expect(importer.doCreate(physicalId, doCreateProps))
+
+        // THEN
+          .resolves.toEqual({ CertificateArn: certArn });
+        expect(getItemStub.calledOnce).toBe(true);
+        expect(getCertificateFake.calledOnce).toBe(true);
+        expect(importCertificateFake.calledOnce).toBe(true);
```

Comment:
I'm probably missing something, but I am confused by this assertion. Why do we import a cert if it already exists in ACM?

Reply:
Ah thanks for clarifying that. Can we add a comment or rename the test to help clarify this?